src/main/java/pl/javastart/voting/VotingResult.java

```diff
@@ -1,27 +1,46 @@
 package pl.javastart.voting;
 
-/**
- * Możesz dodać kolejne metody i pola do klasy. Nie zmieniaj istniejących metod.
- */
+import java.util.List;
+
 public class VotingResult {
+    private List<Vote> voteList;
+
+    public VotingResult(List<Vote> voteList) {
+        this.voteList = voteList;
+    }
 
-    /**
-     * Metoda powinna drukować wyniki głosowania w takiej postaci:
-     * Głosów za: 56.53%
-     * Głosów przeciw: 35.00%
-     * Wstrzymało się: 8.47%
-     */
     public void printResults() {
-        // metoda powinna drukować wyniki głosowania
+        double voteFor = 0;
+        double voteAgains = 0;
+        double withheld = 0;
+        for (Vote vote : voteList) {
+            if (Boolean.TRUE.equals(vote.getVote())) {
+                voteFor++;
+            } else if (Boolean.FALSE.equals(vote.getVote())) {
+                voteAgains++;
+            } else {
+                withheld++;
+            }
+        }
+        System.out.println("Wstrzymało się: " + String.format("%.2f", (withheld / voteList.size()) * 100) + "%");
+        System.out.println("Głosów za: " + String.format("%.2f", (voteFor / voteList.size()) * 100) + "%");
+        System.out.println("Głosów przeciw: " + String.format("%.2f", (voteAgains / voteList.size()) * 100) + "%\n");
     }
 
-    /**
-     * Metoda przyjmuje imię i nazwisko głosującego, np "Zigniew Siobro".
-     * Uzupełnij tę metodę, żeby drukowała informację w formie:
-     * Zigniew Siobro: ZA
-     * Nie zmieniaj sygnatury tej metody!
-     */
     public void printVoteForVoter(String voterName) {
-
+        for (Vote vote : voteList) {
+            if (voterName.equals(vote.getVoter())) {
+                try {
+                    boolean voteResult = vote.getVote();
+                    if (voteResult) {
+                        System.out.println(voterName + ": ZA");
+                    } else {
+                        System.out.println(voterName + ": PRZECIW");
+                    }
+                } catch (NullPointerException e) {
+                    System.out.println(voterName + ": WSTRZYMAŁ SIĘ");
+                }
```

Comment:
Nie łapiemy NullPointerów! Zamiast tego dodaj if sprawdzający czy vote nie jest nullem. W tym przypadku zapis:
`boolean voteResult = vote.getVote();` powoduje ten null pointer. Gdybyś zapisał: `Boolean voteResult = vote.getVote();`, to nullpointera by nie było i możesz taką zmienną porównąć do nulla.